### PR TITLE
Added void null and empty string argument checks

### PR DESCRIPTION
**Pull request by Thejacques1**
- Added void null and empty string argument checks. 
- Added Assert that takes an action to deal to void methods

### CODE REVIEW

**Comment by Thejacques1** [comment]
I can't run the tests on my pc at this stage. Please run them for me. 🙉

File: Source/Ensure.Net/Ensure.cs [hunk 1]
```diff
@@ -29,10 +33,38 @@ public static class Ensure
...
+        /// </summary>
+        /// <param name="value">The string variable to be checked.</param>
+        public static void NotNullOrEmpty(string value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException("Variable cannot be null.");
+            }
+
+            if (string.IsNullOrEmpty(value))
```
Comment (bernarden) [inline Source/Ensure.Net/Ensure.cs]: Check for empty string instead.

File: Source/Ensure.Net/Ensure.cs [hunk 2]
```diff
@@ -42,29 +43,39 @@ public static class Ensure
...
         }
         /// <summary>
         /// Determine whether a string is null, if so throws an ArgumentNullException
         /// Determine whether a string is empty, if so throws an ArgumentException
         /// </summary>
         /// <param name="value">The string variable to be checked.</param>
-        public static void NotNullOrEmpty(string value)
+        public static void NotNullOrEmpty(string value, string parameterName)
         {
+            if (string.IsNullOrEmpty(parameterName))
```
Comment (bernarden) [inline Source/Ensure.Net/Ensure.cs]: We should do either this check or the two checks bellow.  I think the checks below are the ones to keep.

File: Source/Ensure.Net/Ensure.cs [hunk 3]
```diff
@@ -42,29 +43,39 @@ public static class Ensure
         /// Determine whether an object is null, if so throws a ArgumentNullException
         /// </summary>
         /// <param name="value">The object to be checked.</param>
-        public static void NotNull(object value)
+        public static void NotNull(object value, string parameterName)
```
Comment (bernarden) [inline Source/Ensure.Net/Ensure.cs]: Add comment for the new param.

File: Source/Ensure.Net/Ensure.cs [hunk 4]
```diff
@@ -42,29 +43,39 @@ public static class Ensure
...
+                throw new ArgumentNullException("", "Variable cannot be null.");
             }
         }
         /// <summary>
         /// Determine whether a string is null, if so throws an ArgumentNullException
         /// Determine whether a string is empty, if so throws an ArgumentException
         /// </summary>
         /// <param name="value">The string variable to be checked.</param>
-        public static void NotNullOrEmpty(string value)
+        public static void NotNullOrEmpty(string value, string parameterName)
```
Comment (bernarden) [inline Source/Ensure.Net/Ensure.cs]: Add comment for the new param.

File: Source/Ensure.Net.Tests/EnsureTest.cs [hunk 5]
```diff
@@ -27,20 +27,28 @@ public void NotNullCheckShouldThrowArgumentNullExceptionIfVariableIsNull()
...
             Assert.Equal("Variable cannot be null.", ex.Message);
         }
 
         [Test]
+        public void NotNullCheckShouldNotThrowArgumentNullExceptionIfVariableIsNotNull()
+        {
+            string variableName = "Test";
+
+            Ensure.NotNull(variableName, nameof(variableName));
+        }
```
Comment (bernarden) [inline Source/Ensure.Net.Tests/EnsureTest.cs]: Maybe do an assertion on the output.

**Review by bernarden — APPROVED** [review]

**Pull request merged**